pkg/controller/controller.go

```diff
@@ -67,6 +67,17 @@ type Controller interface {
 // New returns a new Controller registered with the Manager.  The Manager will ensure that shared Caches have
 // been synced before the Controller is Started.
 func New(name string, mgr manager.Manager, options Options) (Controller, error) {
+	c, err := Configure(name, mgr, options)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add the controller as a Manager components
+	return c, mgr.Add(c)
+}
+
+// Configure a new controller without starting it or adding it to the manager.
+func Configure(name string, mgr manager.Manager, options Options) (Controller, error) {
```

Comment:
Does it seem odd to have a required field in an options parameter (`options.Reconciler`)?

Reply:
@devigned It does to me, but this is how the existing `controller.New` function works. Given that I expect most folks will continue to use `controller.New` I'd prefer to mimic that function's API rather than try to fix it.

Reply:
Not convinced by the name, we now have `Configure` func that returns a new `Controller`.  IMHO whatever the name, it should start with `New`, maybe `NewWithoutManager`?

Also, the godoc here should mention the usecase.

Reply:
Ideally I think this function would be called `New` and the function currently named `New` would be renamed to something more like `AddToManager`. I agree `Configure` is not the best name, but `NewWithoutManager` doesn't seem great either. What does "without manager" mean? It still accepts and uses a manager, it just doesn't add the controller to it.

Perhaps `NewStandalone` or something that aims to represent that the controller runs without being added to the manager?

Reply:
`NewUnmanaged` maybe?